client/web/compose/src/components/ModuleFields/Editor/Record.vue

```diff
@@ -506,10 +504,26 @@ export default {
       this.filter.pageCursor = next ? this.filter.nextPage : this.filter.prevPage
     },
 
-    getOptionKey ({ value }) {
+    findRecord (recordID) {
+      return this.options.find(r => r.value === recordID)
+    },
+
+    getOptionKey (v) {
+      if (typeof v === 'string') {
+        return this.options.find(r => r.value === v).value
+      }
+      const { value } = v || {}
       return value
     },
 
+    getOptionLabel (v) {
+      if (typeof v === 'string') {
+        return this.findRecord(v).label
```

Comment:
Same here ->Whenever you do a .value, you need to be sure that what you're reading from is always an object and can never be undefined